async-raft/src/core/append_entries.rs

```diff
@@ -242,15 +242,13 @@ impl<D: AppData, R: AppDataResponse, N: RaftNetwork<D>, S: RaftStorage<D, R>> Ra
                 }
             })
             .collect();
-        // If we actually have some cached entries to apply, then we optimistically update, as
-        // `self.last_applied` is held in-memory only, and if an error does come up, then
-        // Raft will go into shutdown.
-        if let Some(index) = last_entry_seen {
-            self.last_applied = index;
-            self.report_metrics();
-        }
+
         // If we have no data entries to apply, then do nothing.
         if entries.is_empty() {
+            if let Some(index) = last_entry_seen {
+                self.last_applied = index;
+                self.report_metrics();
+            }
```

Comment:
Actually, yes. Keep it. That's my bad, we do want to keep that code because we may still actually have a new `last_applied` value in the case of empty payloads and such.